pkg/cloud/services/ec2/instances.go

```diff
@@ -178,7 +179,11 @@ func (s *Service) CreateInstance(scope *scope.MachineScope) (*infrav1.Instance,
 	}
 
 	// Set userdata.
-	input.UserData = aws.String(*scope.Machine.Spec.Bootstrap.Data)
+	userData, err := scope.GetBootstrapData()
+	if err != nil {
+		return nil, err
```

Comment:
Sorry for the additional comment, but I think this error will get "swallowed" in that it's only written to the pod log, and it's never visible to the user anywhere. We generate events in this method (CreateInstance), but only for certain failures. Maybe we should merge this as-is, and have a follow-up issue to adjust the success/fail event creation in CreateInstance so it covers all returnable errors?

Reply:
Added a error and success event for creation